pkg/grpc/authn/tokenbased/extractor.go

```diff
@@ -115,7 +115,12 @@ func (e *extractor) withRoleNames(ctx context.Context, token *tokens.TokenInfo,
 		authProvider:  authProvider,
 	}
 	if id.friendlyName == "" {
-		id.friendlyName = fmt.Sprintf("anonymous bearer token with roles %s (expires %v)", strings.Join(roleNames, ","), token.Expiry())
+		// Note we use roles as seen in the token, without filtering.
+		id.friendlyName = fmt.Sprintf("anonymous bearer token %q with roles [%s] (jti: %s, expires: %s)",
+			token.Name,
+			strings.Join(roleNames, ","),
+			token.ID,
+			token.Expiry().Format(time.RFC3339))
```

Comment:
Does it make sense to include the time zone shift?

Reply:
I don't think it's important either way, so the default is best.